src/Symfony/Component/Security/Http/Authenticator/AccessToken/AccessTokenAuthenticator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\AccessToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\AccessTokenCredentials;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+class AccessTokenAuthenticator implements AuthenticatorInterface
+{
+    private AuthenticationFailureHandlerInterface $failureHandler;
+
+    /**
+     * @param AccessTokenExtractorInterface[] $accessTokenExtractors
+     */
+    public function __construct(
+        private readonly UserProviderInterface $userProvider,
+        private readonly AccessTokenHandlerInterface $accessTokenHandler,
+        private readonly iterable $accessTokenExtractors,
```

Comment:
Maybe could we simplify the public API of the component by accepting only a single instance of `AccessTokenExtractorInterface` here, and introducing a `ChainAccessTokenExtractor` that will loop over the registered extractors when using the fullstack framework?

We usually do that in the other components.

Reply:
I think both options are fine personally.

But to counter your example @Spomky (and huge +1 for thinking about DX!): we can keep the config `string|string[]` and automatically wire the chain extractor when an array was given.

Reply:
Indeed I would keep the current config or do what @wouterj suggests.

Reply:
I haven't though on that possibility.
In the last commit, I changed the firewall config and it is now possible to set a string or a list of strings.
The factory will directly inject the extractor if only one is set, otherwise it will create a service definition based on the new `ChainAccessTokenExtractor` and inject it.